contracts/finality/src/contract.rs

```diff
@@ -313,32 +321,300 @@ pub(crate) mod tests {
     }
 
     #[test]
-    fn test_update_admin_invalid_address() {
+    fn test_admin_update_rejects_malformed_addresses() {
         let mut deps = mock_deps_babylon();
         let init_admin = deps.api.addr_make(INIT_ADMIN);
-        let invalid_new_admin = "invalid-new-admin";
         let bsn_id = "op-stack-l2-11155420".to_string();
-        let min_pub_rand = get_random_u64_range(1, 1000000);
+        let min_pub_rand = 100;
 
+        // Initialize contract
         let instantiate_msg = InstantiateMsg {
             admin: init_admin.to_string(),
             bsn_id,
             min_pub_rand,
         };
+        let info = message_info(&deps.api.addr_make(CREATOR), &[]);
+        let res = instantiate(deps.as_mut(), mock_env(), info, instantiate_msg).unwrap();
+        // Assert that no messages were sent
+        assert_eq!(0, res.messages.len());
+        // Use assert_admin to verify that the admin was set correctly
+        ADMIN.assert_admin(deps.as_ref(), &init_admin).unwrap();
+
+        let admin_info = message_info(&init_admin, &[]);
+
+        // Test various invalid address formats
+        let invalid_addresses = vec![
```

Comment:
I think the function for validating address is valid should have been tested by CosmWasm libraries and we don't need to do it again here